docs/troubleshoot/app-framework/app-troubleshoot.md

```diff
@@ -1,8 +1,194 @@
+
+
+## Application plug-in not in Zowe Desktop
+
+**Symptom:**  
+An application plug-in is not appearing in the Zowe Desktop.
+
+**Troubleshooting:**   
+To check if the plug-in loaded successfully, enter the following URL in a browser to display all successfully loaded Zowe plug-ins:
+
+`https://my.mainframe.com:8544/plugins?type=application`
+
+You can also search the [node server logs](app-mustgather.md) for the plug-in identifier, for example `org.zowe.sample.app`. If the plug-in loaded successfully, you will find the following message:
+```
+[2019-08-06 13:54:21.341 _zsf.bootstrap INFO] - Plugin org.zowe.sampleapp at path=zlux\org.zowe.sampleapp loaded.
+```
+If the plug-in did not load successfully, you will find the following message:
+```
+[2019-08-06 13:54:21.208 _zsf.bootstrap WARNING] - Error: org.zowe.sampleapp 
+```
+If the identifier is not in the logs, make sure the plug-in's locator file is in the `/zlux-app-server/deploy/instance/ZLUX/plugins/` directory. The plug-in locator is a `.json` file, usually with same name as the identifier, for example `org.zowe.sampleapp.json`. Open the file and make sure that the path defined with the `pluginLocation` attribute is correct. If the path is relative, make sure it is relative to the `zlux-app-server/bin` directory. 
+
+For more information on loading plug-ins to the Desktop, see [Adding Your App to the Desktop](https://github.com/zowe/workshop-user-browser-app/blob/master/README.md).
+
+
+## Error: You must specify MVD_DESKTOP_DIR in your environment
+
+**Symptom:** 
+
+A plug-in build in your local environment using `npm run start` or `npm run build` failed with an error message about a missing MVD_DESKTOP_DIR environment variable. 
+
+**Solution:**   
+Add the Zowe Desktop directory path to the MVD_DESKTOP_DIR environment variable. To specify the path, run the following commands in your Windows console or Linux bash shell:
+
+- Windows
+  ```
+  export MVD_DESKTOP_DIR=<zlux-root-dir>/zlux-app-manager/virtual-desktop
+  ```
+
+- Mac Os/Linux
+  ```
+  set MVD_DESKTOP_DIR=<zlux-root-dir>/zlux-app-manager/virtual-desktop
+  ```
+
+## Error: Problem making eureka request { Error: connect ECONNREFUSED }
+
+**Symptom:** 
+
+The Zowe started task `ZWESVSTC` log contains error messages reporting problems connecting 
+
+```
+Problem making eureka request { Error: connect ECONNREFUSED 10.1.1.2:7553
+at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1195:14)
+errno: 'ECONNREFUSED',
+code: 'ECONNREFUSED',
+syscall: 'connect',
+address: '10.1.1.2',
+port: 7553 }
+```
+
+**Solution:**   
+These messages are timing related where different Eureka servers are coming up and trying to connect to each other and warning that the endpoint they are trying to make handshake with is not available.  When all of the Eurka services have started, these errors will stop being logged.  

```

Comment:
Pedant says "timing related" should be hyphenated because it's an adjective.  And "make handshake" sounds better without "make", or change "make handshake" to "perform a handshake".